File: app/src/main/java/net/osdn/gokigen/a01d/camera/panasonic/wrapper/IPanasonicCamera.java
package net.osdn.gokigen.a01d.camera.panasonic.wrapper;

import java.util.List;

public interface IPanasonicCamera
{
    boolean hasApiService(String serviceName);
    List<IPanasonicApiService> getApiServices();

    String getFriendlyName();
    String getModelName();
    String getddUrl();
    String getCmdUrl();
    String getObjUrl();
    String getPictureUrl();
}

File: app/src/main/java/net/osdn/gokigen/a01d/camera/panasonic/wrapper/IPanasonicCamera.kt
package net.osdn.gokigen.a01d.camera.panasonic.wrapper

interface IPanasonicCamera {
    fun hasApiService(serviceName: String?): Boolean
    fun getApiServices(): List<IPanasonicApiService?>?

    fun getFriendlyName(): String?
    fun getModelName(): String?

    fun getDdUrl(): String?
    fun getCmdUrl(): String?
    fun getObjUrl(): String?
    fun getPictureUrl(): String?

    fun getClientDeviceUuId(): String

    fun getCommunicationSessionId() : String?
    fun setCommunicationSessionId(sessionId: String?)
}